packages/firebase_analytics/firebase_analytics_web/lib/firebase_analytics_web.dart

```diff
@@ -44,8 +44,7 @@ class FirebaseAnalyticsWeb extends FirebaseAnalyticsPlatform {
   }
 
   @override
-  Future<int?> getSessionId() {
-    // TODO: change UnimplementedError to UnsupportedError
```

Comment:
Sounds like this would warrant a regression test

Reply:
@rrousselGit - I don't think it is necessary. The API is a Future as far as the user is concerned: https://github.com/firebase/flutterfire/blob/master/packages/firebase_analytics/firebase_analytics/lib/src/firebase_analytics.dart#L80

This change fixes that bug. It also resolves to an UnimplementedError on web so it will not be used by any users. The fact that no user has pointed out this incorrect API infers that it isn't being used which makes sense when it does nothing for their web app.

Reply:
This is a behaviour change though (throwing synchronously vs asynchronously). If the behaviour changed, a test helps prevent inadvertently reverting back to the old behaviour. So I think it makes sense

Reply:
"a test helps prevent inadvertently reverting back to the old behaviour" - The test now works (It has never worked but the web e2e tests were always "successful" even when they failed). Now web e2e tests have been fixed, it will fail if behaviour reverts back to previous.

Reply:
Oh so it fixes an existing test. Nevermind then

Reply:
Sorry, I didn't make that clear, it fixes this test: https://github.com/firebase/flutterfire/blob/master/tests/integration_test/firebase_analytics/firebase_analytics_e2e_test.dart#L28 😄